### PR TITLE
Implement builtin benchmarks.

### PR DESCRIPTION
**Pull request by data-man**
For testing:
`test_benchmarks.c3`
```
import std::io;

fn void! bench1() @benchmark
{
}

fn void! bench2() @benchmark
{
}
```
`$ c3c compile-benchmark test_benchmarks.c3`

### CODE REVIEW

File: src/compiler/sema_decls.c [hunk 1]
```diff
@@ -2593,7 +2608,7 @@ static inline bool sema_analyse_func(SemaContext *context, Decl *decl, bool *era
 		}
 		if (decl->name == kw_main)
 		{
-			if (is_test) SEMA_ERROR(decl, "Main functions may not be annotated @test.");
+			if (is_test || is_benchmark) SEMA_ERROR(decl, "Main functions may not be annotated @test and @benchmark.");
```
Comment (lerno) [inline src/compiler/sema_decls.c:2611]: Should be "nor" rather than "and"

Comment (data-man) [reply src/compiler/sema_decls.c:2611]: This is "or".

Comment (data-man) [reply src/compiler/sema_decls.c:2611]:
```
fn void! main() @test // or @benchmark
{
}
```
`$ c3c compile-test test_bad.c3` // or `c3c compile-benchmark test_bad.c3`
```
1: fn void! main() @test // or @benchmark
            ^^^^
(.../test_bad.c3:1:10) Error: Main functions may not be annotated @test and @benchmark.
```

Comment (data-man) [reply src/compiler/sema_decls.c]: Oh, you wrote about an error messages! Fixed, I hope.

Comment (lerno) [reply src/compiler/sema_decls.c]: Yes, the message :D

**Comment by lerno** [comment]
Thank you.

**Pull request merged**